### PR TITLE
[SC-650] Rate limit integration tests

### PR DESCRIPTION
**Pull request by hexonaut**
Adding rate limit integration tests. Holding off on CCTP ones until https://github.com/marsfoundation/spark-alm-controller/pull/19 is merged.

### CODE REVIEW

File: test/mainnet-fork/PsmCalls.t.sol [hunk 1]
```diff
@@ -67,6 +67,43 @@ contract MainnetControllerSwapUSDSToUSDCTests is ForkTestBase {
         assertEq(dai.allowance(address(almProxy),  address(PSM)),     0);
     }
 
+    function test_swapUSDSToUSDC_rateLimited() external {
+        bytes32 key = mainnetController.LIMIT_USDS_TO_USDC();
+        vm.startPrank(relayer);
+
+        // Rate limited so split this in two
```
Comment (barrutko) [inline test/mainnet-fork/PsmCalls.t.sol:74]: I'd rather make minting unlimited for this test

File: test/mainnet-fork/PsmCalls.t.sol [hunk 2]
```diff
@@ -132,5 +169,31 @@ contract MainnetControllerSwapUSDCToUSDSTests is ForkTestBase {
         assertEq(dai.allowance(address(almProxy),  address(PSM)),     0);
     }
 
+    function test_swapUSDCToUSDS_rateLimited() external {
+        bytes32 key = mainnetController.LIMIT_USDS_TO_USDC();
+        vm.startPrank(relayer);
+
+        // Rate limited so split this in two
```
Comment (barrutko) [inline test/mainnet-fork/PsmCalls.t.sol:176]: Is this comment relevant here?

File: test/mainnet-fork/PsmCalls.t.sol (hunk 1 above)
Comment (hexonaut) [reply test/mainnet-fork/PsmCalls.t.sol:74]: Not sure I understand.

Comment (hexonaut) [reply test/mainnet-fork/PsmCalls.t.sol:74]: Sure

File: test/mainnet-fork/PsmCalls.t.sol [hunk 3]
```diff
@@ -132,5 +170,30 @@ contract MainnetControllerSwapUSDCToUSDSTests is ForkTestBase {
         assertEq(dai.allowance(address(almProxy),  address(PSM)),     0);
     }
 
+    function test_swapUSDCToUSDS_rateLimited() external {
+        bytes32 key = mainnetController.LIMIT_USDS_TO_USDC();
```
Comment (lucas-manuel) [inline test/mainnet-fork/PsmCalls.t.sol:174]: `LIMIT_USDC_TO_USDS` and the rate limits should decrease

Comment (lucas-manuel) [reply test/mainnet-fork/PsmCalls.t.sol:174]: Should also follow exact same structure as `test_swapUSDSToUSDC_rateLimited`

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

**Review by lucas-manuel — APPROVED** [review]

**Review by lucas-manuel — APPROVED** [review]

**Pull request merged**